### PR TITLE
New sub_test feature: testfile.compenv

### PR DESCRIPTION
**Pull request by ben-albrecht**
This PR allows a `testfile.compenv` in tests that will modify the environment during compilation in the same way `testfile.execenv` modifies the environment during execution.

Because of the similarities, the implementation of `.compenv` in sub_test closely mirrors that of `.execenv.`

This feature was motivated by the need to test compiler flags overriding `CHPL_*` environment variables (coming in a near future PR).

Confirmed that the changes work via manual tests.

[`Paratested √`]

### CODE REVIEW

**Comment by ben-albrecht** [comment]
Maybe `sub_test` enthusiasts (@Kyle-B & @ronawho) might be interested?

File: util/test/sub_test [hunk 1]
```diff
@@ -1013,6 +1023,7 @@ for testname in testsrc:
             suffix=='.compopts' or suffix=='.perfcompopts' or
             suffix=='.chpldocopts' or
             suffix=='.execenv' or suffix=='.perfexecenv' or
+            suffix=='.compenv' or suffix=='.compexecenv' or
```
Comment (Kyle-B) [inline util/test/sub_test]: Why is `.compeecenv` here?

Comment (ben-albrecht) [inline util/test/sub_test]: Yep!

File: util/test/sub_test [hunk 2]
```diff
@@ -1205,6 +1221,14 @@ for testname in testsrc:
             testenv[tev.split('=')[0].strip()] = tev.split('=')[1].strip()
         del tev
 
+    testcompenv = {}
+    if len(globalCompenv)!=0 or len(compenv)!=0:
+        for cev in globalCompenv:
+            testcompenv[cev.split('=')[0].strip()] = cev.split('=')[1].strip()
```
Comment (Kyle-B) [inline util/test/sub_test]: How about:

```python
for var, val in [cev.split('=') in globalCompenv]:
    testcompenv[var.strip()] = val.strip()
```

Comment (ben-albrecht) [inline util/test/sub_test]: Sure - should I improve the testenv code above as well then?

Comment (Kyle-B) [inline util/test/sub_test]: That would be nice I think.

**Pull request merged**